feature-libs/product-configurator/rulebased/components/conflict-solver-dialog/configurator-conflict-solver-dialog.component.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { ConfiguratorConflictSolverDialogComponent } from './configurator-conflict-solver-dialog.component';
+import { I18nTestingModule, LanguageService } from '@spartacus/core';
+import {
+  ICON_TYPE,
+  IconLoaderService,
+  IconModule,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import {
+  ConfigFormUpdateEvent,
+  Configurator,
+  ConfiguratorCommonsService,
+} from '@spartacus/product-configurator/rulebased';
+import { Type } from '@angular/core';
+
+export class MockIconFontLoaderService {
+  getStyleClasses(_iconType: ICON_TYPE): string {
+    return 'fas fa-exclamation-circle';
+  }
+  addLinkResource() {}
+  getHtml(_iconType: ICON_TYPE) {}
+  getFlipDirection(): void {}
+}
+
+class MockConfiguratorCommonsService {
+  updateConfiguration(): void {}
+}
+
+class MockLaunchDialogService implements Partial<LaunchDialogService> {
+  get data$(): Observable<any> {
+    return of({
+      conflictGroup: of([]),
+      routerData: of(undefined),
+    });
+  }
+
+  closeDialog(_reason: string): void {}
+}
+
+describe('ConfiguratorConflictSolverDialogComponent', () => {
+  let component: ConfiguratorConflictSolverDialogComponent;
+  let fixture: ComponentFixture<ConfiguratorConflictSolverDialogComponent>;
+  let configuratorCommonsService: ConfiguratorCommonsService;
+  let launchDialogService: LaunchDialogService;
+  let mockLanguageService;
+
+  beforeEach(
+    waitForAsync(() => {
+      mockLanguageService = {
+        getAll: () => of([]),
+        getActive: jasmine.createSpy().and.returnValue(of('en')),
+      };
+
+      TestBed.configureTestingModule({
+        imports: [I18nTestingModule, IconModule],
+        declarations: [ConfiguratorConflictSolverDialogComponent],
+        providers: [
+          { provide: IconLoaderService, useClass: MockIconFontLoaderService },
+          { provide: LanguageService, useValue: mockLanguageService },
+          {
+            provide: ConfiguratorCommonsService,
+            useClass: MockConfiguratorCommonsService,
+          },
+          { provide: LaunchDialogService, useClass: MockLaunchDialogService },
+        ],
+      }).compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      ConfiguratorConflictSolverDialogComponent
+    );
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+
+    configuratorCommonsService = TestBed.inject(
+      ConfiguratorCommonsService as Type<ConfiguratorCommonsService>
+    );
+
+    spyOn(configuratorCommonsService, 'updateConfiguration').and.callThrough();
+
+    launchDialogService = TestBed.inject(LaunchDialogService);
+    spyOn(launchDialogService, 'closeDialog').and.callThrough();
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
```

Comment:
as indicated above some components tests are missing, so for example one could assert
- if we have a configuration without conflicts the modal is not rendered
- if we have a configuration with one conflict, the modal is rendered and the conflict is visualized using the expected components depending on UI Type (ideally each UI Type is covered)
-  if we have a configuration with more than one conflict, only the first one is visualized

Reply:
@Uli-Tiger, the first point is not valid according to my point of view because the launching of the modal depends on the dialog event listener. The listener will not open the modal if there are no conflicts, that's why it is overengineered to write a test for it.

Reply:
agree